### PR TITLE
Form Elements

### PR DESCRIPTION
**Pull request by richmarisa**
Add a form elements directory to contain various form element blade templates.

**This PR does the following**

- Change the name of a directory under views/components from cwd-framework to cd.
- Add a form subdirectory to views/components/cd
- Add test.blade.php and form-item.blade.php to the form subdirectory to begin a set of form elements
- Adjustments to various files to reflect the change from cwd-framework to cd. 

**To Review:**

- [ ] require the laravel starter kit in a new Laravel 10 project. 
- [ ] run php artisan starterkit:install
- [ ] run ./vendor/bin/phpunit to verify that views/components/cd/form/form-item.blade.php was published in the project.

### CODE REVIEW

File: resources/views/cd-index.blade.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<x-cwd-framework.layout.app title=":project_name" subtitle="Custom Development">
+<x-cd.layout.app title="Laraveltest" subtitle="Custom Development">
```
Comment (woodseowl) [inline resources/views/cd-index.blade.php:1]: Not sure how this happened, but the title should be left as a token:
```suggestion
<x-cd.layout.app title=":project_name" subtitle="Custom Development">
```

File: resources/views/components/cd/layout/app.blade.php [hunk 2]
```diff
@@ -7,24 +7,24 @@
 
     <title>{{ $title }}</title>
 
-    <link rel="icon" href="{{ asset('cwd-framework/favicon.ico') }}" type="image/vnd.microsoft.icon"/>
+    <link rel="icon" href="{{ asset('cd/favicon.ico') }}" type="image/vnd.microsoft.icon"/>
```
Comment (woodseowl) [inline resources/views/components/cd/layout/app.blade.php:10]: I wonder if we should split all of these items that are published as assets (that is, in /public) and still deliver them as cwd-framework. I feel like that would be more accurate and clear about what is coming from the outside theme. 

To be clear: we would keep the view components in our short-named directory (/resources/views/components/cd)

File: src/StarterKitServiceProvider.php [hunk 3]
```diff
@@ -13,7 +13,7 @@ class StarterKitServiceProvider extends PackageServiceProvider
 {
     const PACKAGE_NAME = 'starterkit';
 
-    const THEME_NAME = 'cwd-framework';
+    const THEME_NAME = 'cd';
```
Comment (woodseowl) [inline src/StarterKitServiceProvider.php:16]: Per my comment above, we could perhaps add a constant for the components:
```
const COMPONENT_DIR = 'cd';
```
and then use that for the files that are published to the resources directory. Then restore the THEME_NAME value to the theme we are supplying: 'cwd-framework'.

**Review by woodseowl — COMMENTED** [review]
Curious for thoughts on my comments here. (Also one small fix.)

File: resources/views/cd-index.blade.php (hunk 1 above)
Comment (richmarisa) [reply resources/views/cd-index.blade.php:1]: I remember being prompted for the project name at one point.

File: resources/views/components/cd/layout/app.blade.php (hunk 2 above)
Comment (richmarisa) [reply resources/views/components/cd/layout/app.blade.php:10]: I think those assets in /public should remain under cwd-framework

File: src/StarterKitServiceProvider.php (hunk 3 above)
Comment (richmarisa) [reply src/StarterKitServiceProvider.php:16]: What is the theme name used for other than the directory name?  Maybe we don't need THEME_NAME if we use COMPONENT_DIR.

**Comment by woodseowl** [comment]
I've updated the naming and install process per our conversation. I think I've got them all, but I still need to go through the actual install step to test on a project. If that's good to go, I think this looks good and matches what we discussed in the meeting.

**Review by woodseowl — APPROVED** [review]
Tested on the Laravel demo site and all looks good!

File: resources/views/components/cd/form/form-item.blade.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+@php
+    $is_required = $required ?? true;
```
Comment (inaydich) [inline resources/views/components/cd/form/form-item.blade.php:2]: Will it work if $required is not specified? 
Also how do you see form-item component is used?

Comment (richmarisa) [reply resources/views/components/cd/form/form-item.blade.php:2]: Yes, the null coallessing operator is providing a default.  This PR is really just about making sure that the components can be integrated properly; the next update I have, which includes more components, will make the default false, as you had suggested.

**Review by inaydich — APPROVED** [review]
It looks good. Thanks!

**Pull request merged**